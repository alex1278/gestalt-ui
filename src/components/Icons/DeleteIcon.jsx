import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '@material-ui/icons/DeleteOutline';

const DeleteOutlineIconStyle = styled(Icon)`
  font-size: ${props => `${props.size}px !important`};
  color: ${props => `${props.color}`};
`;

const DeleteOutlineIcon = ({ size, color, ...rest }) => <DeleteOutlineIconStyle size={size} color={color} {...rest} />;

DeleteOutlineIcon.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};

DeleteOutlineIcon.defaultProps = {
  size: 24,
  color: 'action',
};

export default DeleteOutlineIcon;
