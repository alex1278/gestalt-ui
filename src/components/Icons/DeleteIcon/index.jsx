import React from 'react';
import styled from 'styled-components';
import FontIcon from 'react-md/lib/FontIcons';

const Icon = styled(FontIcon)`
  // color: ${props => props.theme.removeIconColor} !important;
`;

const DeleteIcon = () => <Icon>delete_sweep</Icon>;

export default DeleteIcon;
