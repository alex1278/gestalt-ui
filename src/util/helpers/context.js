/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
