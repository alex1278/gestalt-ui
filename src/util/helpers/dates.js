import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
