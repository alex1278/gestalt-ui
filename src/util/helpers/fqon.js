// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
