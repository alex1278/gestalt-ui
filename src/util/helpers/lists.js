export function insertItem(array, payload) {
  return [...array, payload];
}

export function removeItem(array, payload) {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a === payload), 1);

  return newArray;
}

export function removeItemById(array, id, keyField = 'id') {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a[keyField] === id), 1);

  return newArray;
}

export function updateItem(array, payload) {
  return array.map((item) => {
    if (item.id === payload.id) {
      return {
        ...payload,
      };
    }

    // Leave every other item unchanged
    return item;
  });
}
