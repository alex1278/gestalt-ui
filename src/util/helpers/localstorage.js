/**
 * getItem
 * @param {*} key
 */
export const getItem = (key) => {
  try {
    return localStorage.getItem(key);
  } catch (err) {
    return undefined;
  }
};

/**
 * saveItem
 * @param {*} key
 * @param {*} value
 */
export const saveItem = (key, value) => {
  try {
    localStorage.setItem(key, value);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};

/**
 * removeItem
 * @param {*} key
 * @param {*} value
 */
export const removeItem = (key) => {
  try {
    localStorage.removeItem(key);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};
