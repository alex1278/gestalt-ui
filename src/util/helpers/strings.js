import { startCase, toLower } from 'lodash';

/**
 * getLastFromSplit
 * Parses and returns the last item in a structure like GrandPareng::Parent::Child
 * @param {String} string
 */
export function getLastFromSplit(string, delimiter = '::') {
  if (string) {
    const split = string.split(delimiter);

    return split[split.length - 1];
  }
  return string;
}

/**
 * truncate
 * @param {String} string
 * @param {Number} at
 * @param {String} trailing
 */
export function truncate(string, at, trailing = '...') {
  if (string) {
    if (string.length < at) {
      return string;
    }

    return `${string.substring(0, at).trim()}${trailing}`;
  }

  return string;
}

/**
 * toTitleCase
 * @param {String} string
 */
export function toTitleCase(string) {
  return startCase(toLower(string));
}


// TODO: Will be rmeoved when we have security config
export const checkIfPassword = field =>
  field && field.length && (field.toUpperCase().includes('PASSWORD') || field.toUpperCase().includes('SECRET') || field.toUpperCase().includes('KEY'));

export default {
  getLastFromSplit,
  truncate,
  toTitleCase,
  checkIfPassword
};

export const removeHostFromURL = (url) => {
  if (url) {
    return url.replace(/^.*\/\/[^/]+:?[0-9]?\//i, '');
  }

  return url;
};

export const isBase64 = (str) => {
  try {
    return btoa(atob(str)) === str;
  } catch (err) {
    return false;
  }
};
