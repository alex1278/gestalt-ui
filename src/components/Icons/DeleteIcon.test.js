import React from 'react';
import DeleteIcon from './DeleteIcon';
import { renderWithTheme } from '../../../test/helpers';

describe('(Component) DeleteIcon', () => {
  it('renders component without exploding', () => {
    const { container } = renderWithTheme(<DeleteIcon />);

    expect(container.firstChild).toMatchSnapshot(1);
  });

  test('it mounts without additional props', () => {
    const { container } = renderWithTheme(<DeleteIcon />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
