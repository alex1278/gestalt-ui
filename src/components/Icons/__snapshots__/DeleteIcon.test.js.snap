// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DeleteIcon it mounts without additional props 1`] = `
.c0 {
  font-size: 24px !important;
  color: action;
}

<svg
  aria-hidden="true"
  class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c0"
  focusable="false"
  role="presentation"
  size="24"
  viewBox="0 0 24 24"
>
  <path
    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
  />
  <path
    d="M0 0h24v24H0V0z"
    fill="none"
  />
</svg>
`;

exports[`(Component) DeleteIcon renders component without exploding 1`] = `
.c0 {
  font-size: 24px !important;
  color: action;
}

<svg
  aria-hidden="true"
  class="MuiSvgIcon-root-1 MuiSvgIcon-colorAction-4 c0"
  focusable="false"
  role="presentation"
  size="24"
  viewBox="0 0 24 24"
>
  <path
    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
  />
  <path
    d="M0 0h24v24H0V0z"
    fill="none"
  />
</svg>
`;
